src/mlpack/methods/preprocess/preprocess_imputer_main.cpp

```diff
@@ -88,36 +90,30 @@ int main(int argc, char** argv)
   // print how many mapping exist in each features
   for (size_t i = 0; i < input.n_rows; ++i)
   {
-    Log::Info << info.NumMappings(i) << " mappings in feature "
-        << i << "." << endl;
+    Log::Info << info.NumMappings(i) << " mappings in feature " << i << "."
+        << endl;
   }
 
-
   arma::Mat<double> output(input);
 
 
   Log::Info << "Performing '" << imputeStrategy << "' imputation strategy "
-            << "to feature '" << feature <<"' of '" << inputFile << "'."
-            << endl;
+      << "on feature '" << feature << endl;
 
+  // custom strategy only
   if (imputeStrategy == "custom")
   {
     Log::Info << "Replacing all '" << missingValue << "' with '" << customValue
-              << "'." << endl;
-
-    Imputer<arma::Mat<double>, Mapper, CustomStrategy> impu(info);
-    impu.template Impute<double>(input,
-                                 output,
-                                 missingValue,
-                                 customValue,
-                                 feature);
+        << "'." << endl;
+    Imputer<double, MapperType, CustomImputation<double>> impu(info);
+    impu.Impute(input, output, missingValue, customValue, feature);
   }
   else
   {
-    Log::Info << "Replacing all '" << missingValue << "' with '" << imputeStrategy
-            << "'." << endl;
+    Log::Info << "Replacing all '" << missingValue << "' with '"
+        << imputeStrategy << "'." << endl;
 
-    Imputer<arma::Mat<double>, Mapper, MeanStrategy> impu(info);
+    Imputer<double, MapperType, MeanImputation<double>> impu(info);
```

Comment:
Do you have any plan to support another imputation strategies?I can add them later on by myself